dataactvalidator/interfaces/validatorStagingInterface.py

```diff
@@ -1,77 +1,52 @@
 from dataactcore.models.stagingInterface import StagingInterface
-from dataactcore.models.stagingModels import FieldNameMap
-from dataactcore.utils.responseException import ResponseException
-from dataactcore.utils.statusCode import StatusCode
-from dataactvalidator.interfaces.validatorJobTrackerInterface import ValidatorJobTrackerInterface
-from sqlalchemy import MetaData, Table
-from sqlalchemy.exc import NoSuchTableError
+from dataactcore.models.stagingModels import Appropriation, ObjectClassProgramActivity, AwardFinancial, AwardFinancialAssistance
 
 
 class ValidatorStagingInterface(StagingInterface):
     """ Manages all interaction with the staging database """
 
-    def dropTable(self,table):
+    def insertSubmissionRecordByFileType(self, record, fileType):
+        """Insert a submitted file row into staging.
+
+        Args:
+            record: record to insert (Dict)
+            submissionId: submissionId associated with this record (int)
+            fileType: originating file type of the record (string)
         """
+        if fileType == "award":
+            rec = AwardFinancialAssistance(**record)
+        elif fileType == "award_financial":
+            rec = AwardFinancial(**record)
+        elif fileType == "appropriations":
+            rec = Appropriation(**record)
+        elif fileType == "program_activity":
+            rec = ObjectClassProgramActivity(**record)
+        self.session.add(rec)
+        self.session.commit()
```

Comment:
That's a good point. Putting the `commit` here essentially breaks the recommended practice of [keeping session lifecycles separate from functions that access/manipulate data](http://docs.sqlalchemy.org/en/latest/orm/session_basics.html).

Can you say more about what you mean by batch inserting? Does that mean committing a transaction after some number of inserts and then starting a new one? Trying to map my mental model of databases to the validator parlance.

One easy thing to do would be to move the `session.commit()` to `threadedValidateJob`:
```
if jobType == csv_record_validation:
    runValidation(job, interface)
    session.commit()
elif jobType = validation:
    runCrossValidation(job, interface)
    session.commit()
```

Alternately, we could do shorter transactions by doing a `session.commit()` down in the validation routines every _n_ records.

I'm inclined to try the first option, since nothing else (in theory) should be trying to update a submission's records in a concurrent transaction.

No matter what, we definitely need to run some large files through the validator (will put in a story for that). At this point, we're just speculating what the bottlenecks might be. The indexes will also slow down our insert performance.